packages/babel-cli/src/babel/dir.js

```diff
@@ -43,18 +43,24 @@ module.exports = function (commander, filenames) {
     }
   }
 
+  let filterDotFiles = function (x) {
+    return x[0] === ".";
+  };
+  let copyFilesFilter = () => false;
+
   function handle(filename) {
     if (!fs.existsSync(filename)) return;
 
     let stat = fs.statSync(filename);
-
     if (stat.isDirectory(filename)) {
       let dirname = filename;
-
-      _.each(util.readdir(dirname), function (filename) {
+      if (commander.copyDotFiles) {
+        copyFilesFilter = filterDotFiles;
```

Comment:
Yeah, after talking to @xtuc, I think the intention (which we clearly didn't communicate well enough in the original issue), is for `--copy-dot-files` to simply change the behavior of `--copy-files`, as opposed to it being its own operation. Sorry for the mixup/confusion :(